### DISCUSSION TITLE
Rule to prefer reference instead of inline function parameter

### DISCUSSION THREAD

**Discussion by Armaldio** [eslint/eslint#13919] [Q&A] [answered]
Take this code for example:

```js
this.xxx('yyy', (value) => {
	console.log(value)
});
```

I'd like to enforce the fact of moving the inline callback out, like this

```js
this.callback = (value) => {
	console.log(value)
}
this.xxx('yyy', this.callback);
```

Is there a rule for that ? All my searches lead to nothing

Thanks 👋

**Comment by mdjermanovic** [comment, marked as answer]
Hi @Armaldio!

You can use the `no-restricted-syntax` rule to enforce this:

```js
/*eslint "no-restricted-syntax": [
    "error",
    {
        "selector": "CallExpression > :matches(FunctionExpression, ArrowFunctionExpression).arguments",
        "message": "Don't use inline callbacks."
    }
]*/

this.xxx('yyy', (value) => { // error
	console.log(value)
});
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgXCJuby1yZXN0cmljdGVkLXN5bnRheFwiOiBbXG4gICAgXCJlcnJvclwiLFxuICAgIHtcbiAgICAgICAgXCJzZWxlY3RvclwiOiBcIkNhbGxFeHByZXNzaW9uID4gOm1hdGNoZXMoRnVuY3Rpb25FeHByZXNzaW9uLCBBcnJvd0Z1bmN0aW9uRXhwcmVzc2lvbikuYXJndW1lbnRzXCIsXG4gICAgICAgIFwibWVzc2FnZVwiOiBcIkRvbid0IHVzZSBpbmxpbmUgY2FsbGJhY2tzLlwiXG4gICAgfVxuXSovXG5cbnRoaXMueHh4KCd5eXknLCAodmFsdWUpID0+IHsgLy8gZXJyb3Jcblx0Y29uc29sZS5sb2codmFsdWUpXG59KTtcbiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

**Reply by Armaldio** [reply]
Thanks a lot !
